extensions/GitExtension/FileExplorerGitIntegration/Models/GitExecute.cs

```diff
@@ -42,18 +42,25 @@ public static GitCommandRunnerResultInfo ExecuteGitCommand(string gitApplication
 
                 // Add timeout for 1 minute
                 process.WaitForExit(TimeSpan.FromMinutes(1));
+
+                if (process.ExitCode != 0)
+                {
+                    Log.Error("Execute Git process exited unsuccessfully with exit code {ExitCode}", process.ExitCode);
+                    return new GitCommandRunnerResultInfo(ProviderOperationStatus.Failure, output, "Execute Git process exited unsuccessfully", string.Empty, null, arguments, process.ExitCode);

```

Comment:
NIT: `exited with a failure error code`  "Exit unsuccessfully" tells me that the process was unsuccessful in exiting.